src/sdk/bucketspace_fs.js

```diff
@@ -634,18 +587,15 @@ class BucketSpaceFS extends BucketSpaceSimpleFS {
         try {
             const { name, policy } = params;
             dbg.log0('BucketSpaceFS.put_bucket_policy: Bucket name, policy', name, policy);
-            const bucket_config_path = this._get_bucket_config_path(name);
-            const { data } = await nb_native().fs.readFile(this.fs_context, bucket_config_path);
-            const bucket = JSON.parse(data.toString());
+            const bucket = await this.configFS.get_bucket_by_name(name);
             bucket.s3_policy = policy;
             const bucket_to_validate = _.omitBy(bucket, _.isUndefined);
-            dbg.log2("put_bucket_policy: bucket properties before validate_bucket_schema",
-            bucket_to_validate);
+            dbg.log2('put_bucket_policy: bucket properties before validate_bucket_schema', bucket_to_validate);
             nsfs_schema_utils.validate_bucket_schema(bucket_to_validate);
             await bucket_policy_utils.validate_s3_policy(bucket.s3_policy, bucket.name, async principal =>
-                 this._get_account_by_name(principal));
+                 this.configFS.get_account_by_name_if_exists(principal));
```

Comment:
why "get_account_by_name_if_exists"?

Reply:
because that's what old _get_account_by_name() did - 
```
async _get_account_by_name(name) {
        const account_config_path = this._get_account_config_path(name);
        try {
            await nb_native().fs.stat(this.fs_context, account_config_path);
            return true;
        } catch (err) {
            return false;
        }
    }
```